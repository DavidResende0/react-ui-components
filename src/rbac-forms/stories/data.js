export const groups = [
  {
    value: '1',
    label: 'Hope',
  },
  {
    value: '2',
    label: 'Christa',
  },
  {
    value: '3',
    label: 'Tania',
  },
  {
    value: '4',
    label: 'Mcintyre',
  },
  {
    value: '5',
    label: 'Tonya',
  },
  {
    value: '6',
    label: 'Marcie',
  },
  {
    value: '7',
    label: 'Cara',
  },
  {
    value: '8',
    label: 'Becker',
  },
  {
    value: '9',
    label: 'Foley',
  },
  {
    value: '10',
    label: 'Perkins',
  },
];

export const usersTableColumns = [{
  property: 'fullname',
  label: 'Full Name',
}, {
  property: 'username',
  label: 'Username',
}, {
  property: 'email',
  label: 'E-mail',
}, {
  property: 'currentgroup',
  label: 'Current Group',
}, {
  property: 'role',
  label: 'Role',
}, {
  property: 'lastlogon',
  label: 'Last Logon',
}, {
  property: 'lastlogoff',
  label: 'Last Logoff',
}];

export const usersTableRows = [
  {
    id: 100,
    fullname: 'Adam the first',
    username: 'Admin',
    email: 'email@neco.com',
    currentgroup: 'Some group',
    role: 'Admin role',
    lastlogon: '06/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 101,
    fullname: 'Marge the middle',
    username: 'La transparenza della qualita',
    email: 'email@jandak.com',
    currentgroup: 'Mediocre group',
    role: 'Shot role',
    lastlogon: '01/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 102,
    fullname: 'Zlatko the last',
    username: 'Peligro',
    email: 'email@jandak.com',
    currentgroup: 'Super group',
    role: 'Shot role',
    lastlogon: '08/25/18 06:58:14 UTC',
    lastlogoff: '010/30/18 06:58:14 UTC',
  },
];

export const tags = [{
  name: 'Tag 1',
  value: 'Tag value 1',
}, {
  name: 'Tag 2',
  value: 'Tag value 2',
}, {
  name: 'Tag 3',
  value: 'Tag value 3',
}, {
  name: 'Tag 4',
  value: 'Tag value 4',
}, {
  name: 'Tag 5',
  value: 'Tag value 5',
}];
