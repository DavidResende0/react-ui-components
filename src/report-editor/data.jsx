/* Columns Tab - Available Fields */
export const columnOptions = [{
  value: 'Cloud Manager: Subscription',
  label: 'Cloud Manager: Subscription',
}, {
  value: 'Cloud Manager: Total Cloud Memory',
  label: 'Cloud Manager: Total Cloud Memory',
  type: 'number',
}, {
  value: 'Vms: Date Created',
  label: 'Vms: Date Created',
  type: 'datetime',
}, {
  value: 'Cloud Manager: Last Inventory Date',
  label: 'Cloud Manager: Last Inventory Date',
  type: 'datetime',
}, {
  value: 'Cloud Manager: Last Refresh Date',
  label: 'Cloud Manager: Last Refresh Date',
  type: 'datetime',
}, {
  value: 'Management Events: Date Created',
  label: 'Management Events: Date Created',
  type: 'datetime',
}, {
  value: 'Vms: Last Sync Time',
  label: 'Vms: Last Sync Time',
  type: 'datetime',
}, {
  value: 'CPU Usage Rate Average Avg Over Time Period',
  label: 'CPU Usage Rate Average Avg Over Time Period',
  type: 'number',
}, {
  value: 'Memory Usage Rate Average Avg Over Time Period',
  label: 'Memory Usage Rate Average Avg Over Time Period',
  type: 'number',
}, {
  value: 'Cloud Manager: Total Number of Logical CPUs',
  label: 'Cloud Manager: Total Number of Logical CPUs',
  type: 'number',
}, {
  value: 'Cloud Manager: Total Number of Physical CPUs',
  label: 'Cloud Manager: Total Number of Physical CPUs',
  type: 'number',
}, {
  value: 'Management Events: Container Group Name',
  label: 'Management Events: Container Group Name',
}, {
  value: 'Region Description',
  label: 'Region Description',
}, {
  value: 'Cloud Manager: Hostname',
  label: 'Cloud Manager: Hostname',
}, {
  value: 'Management Events: Destination VM Location',
  label: 'Management Events: Destination VM Location',
}];

/* Columns Tab - Configure Report Columns */
export const baseTheReportOptions = [
  {
    value: 'Availability Zones',
    label: 'Availability Zones',
  },
  {
    value: 'Bottleneck Events',
    label: 'Bottleneck Events',
  },
  {
    value: 'Chargeback for Images',
    label: 'Chargeback for Images',
  },
  {
    value: 'Chargeback for Projects',
    label: 'Chargeback for Projects',
  },
  {
    value: 'Chargeback for Vms',
    label: 'Chargeback for Vms',
  },
  {
    value: 'Cloud Networks',
    label: 'Cloud Networks',
  },
  {
    value: 'Cloud Providers',
    label: 'Cloud Providers',
  },
  {
    value: 'Cloud Resource Quotas',
    label: 'Cloud Resource Quotas',
  },
  {
    value: 'Chargeback for Vms',
    label: 'Chargeback for Vms',
  },
  {
    value: 'Cloud Tenants',
    label: 'Cloud Tenants',
  },
  {
    value: 'Cloud Volumes',
    label: 'Cloud Volumes',
  },
  {
    value: 'Cluster / Deployment Roles',
    label: 'Cluster / Deployment Roles',
  },
  {
    value: 'Compliance Histories',
    label: 'Compliance Histories',
  },
  {
    value: 'Configuration Managers',
    label: 'Configuration Managers',
  },
  {
    value: 'Configured Systems (Foreman)',
    label: 'Configured Systems (Foreman)',
  },
];

/* Columns Tab - Report Creation Timeout */
export const creationTimeoutOptions = [
  {
    value: '1 Hour',
    label: '1 Hour',
  },
  {
    value: '2 Hours',
    label: '2 Hours',
  },
  {
    value: '3 Hours',
    label: '3 Hours',
  },
  {
    value: '4 Hours',
    label: '4 Hours',
  },
  {
    value: '5 Hours',
    label: '5 Hours',
  },
  {
    value: '6 Hours',
    label: '6 Hours',
  },
];

/* Formatting Tab - PDF Output */
export const pdfOptions = [
  {
    value: 'A0 - 841mm x 1189mm',
    label: 'A0 - 841mm x 1189mm',
  },
  {
    value: 'A1 - 594mm x 841mm',
    label: 'A1 - 594mm x 841mm',
  },
  {
    value: 'A2 - 420mm x 594mm',
    label: 'A2 - 420mm x 594mm',
  },
  {
    value: 'A3 - 297mm x 420mm',
    label: 'A3 - 297mm x 420mm',
  },
  {
    value: 'A4 - 210mm x 297mm',
    label: 'A4 - 210mm x 297mm',
  },
  {
    value: 'US Executive - 7.25in x 10.5in',
    label: 'US Executive - 7.25in x 10.5in',
  },
  {
    value: 'US Folio - 8.5in x 13.0in',
    label: 'US Folio - 8.5in x 13.0in',
  },
  {
    value: 'US Government - 8.0in x 11.0in',
    label: 'US Government - 8.0in x 11.0in',
  },
  {
    value: 'US Executive - 17.0in x 11.0in',
    label: 'US Executive - 17.0in x 11.0in',
  },
  {
    value: 'US Legal - 8.5in x 14.0in',
    label: 'US Legal - 8.5in x 14.0in',
  },
];

/* Summary Tab - Sort Order */
export const sortOrderOptions = [
  {
    value: 'Asc',
    label: 'Ascending',
  },
  {
    value: 'Desc',
    label: 'Descending',
  },
];

/* Summary Tab - Sort Breaks */
export const sortBreaksOptions = [
  {
    value: 'No',
    label: 'No',
  },
  {
    value: 'Yes',
    label: 'Yes',
  },
  {
    value: 'Counts',
    label: 'Counts',
  },
];

/* Summary Tab - Number of rows to show */
export const rowNumberOptions = [
  {
    value: 'All',
    label: 'All',
  },
];

/* Summary Tab - Format on Summary Row */
export const rowOptions = [
  {
    value: 'Reset to default',
    label: 'Reset to default',
  },
  {
    value: 'Convert Numbers Larger than 1.0e+15 to Exponential Form',
    label: 'Convert Numbers Larger than 1.0e+15 to Exponential Form',
  },
  {
    value: 'Currency, 2 Decimals ($1,234.00)',
    label: 'Currency, 2 Decimals ($1,234.00)',
  },
  {
    value: 'Number (1,234)',
    label: 'Number (1,234)',
  },
  {
    value: 'Number (1,234.0)',
    label: 'Number (1,234.0)',
  },
];

/* Summary - Time Format on Summary Row */
export const formatSummaryTimeOptions = [
  {
    value: 'Date/Time (M/D/Y H:M:S Z)',
    label: 'Date/Time (M/D/Y H:M:S Z)',
  }, {
    value: 'Date (M/D)',
    label: 'Date (M/D)',
  }, {
    value: 'Date (M/D/YY)',
    label: 'Date (M/D/YY)',
  }, {
    value: 'Date (M/D/YYY)',
    label: 'Date (M/D/YYY)',
  }, {
    value: 'Day Full (Monday)',
    label: 'Day Full (Monday)',
  }, {
    value: 'Day Short (Mon)',
    label: 'Day Short (Mon)',
  }, {
    value: 'Day of Month (27)',
    label: 'Day of Month (27)',
  }, {
    value: 'Day of Month (27th)',
    label: 'Day of Month (27th)',
  }, {
    value: 'Day of Week (1)',
    label: 'Day of Week (1)',
  }, {
    value: 'Hour of Day (24)',
    label: 'Hour of Day (24)',
  }, {
    value: 'Month Full',
    label: 'Month Full',
  }, {
    value: 'Year (YYYY)',
    label: 'Year (YYYY)',
  },
];

/* Summary Tab - Original value to time fields */
export const timeValues = [
  {
    value: 'Original Value',
    label: 'Original Value',
  },
  {
    value: 'Hour',
    label: 'Hour',
  },
  {
    value: 'Day',
    label: 'Day',
  },
  {
    value: 'Week',
    label: 'Week',
  },
  {
    value: 'Month',
    label: 'Month',
  },
  {
    value: 'Quarter',
    label: 'Quarter',
  },
  {
    value: 'Year',
    label: 'Year',
  },
  {
    value: 'Hour of the Day',
    label: 'Hour of the Day',
  },
  {
    value: 'Day of the Week',
    label: 'Day of the Week',
  },
  {
    value: 'Day of the Month',
    label: 'Day of the Month',
  },
  {
    value: 'Week of the Year',
    label: 'Week of the Year',
  },
  {
    value: 'Month of the Year',
    label: 'Month of the Year',
  },
];

/* Summary Tab - Original value to time fields */
export const stringValues = [
  {
    value: 'Human Readable Model Name',
    label: 'Human Readable Model Name',
  }, {
    value: 'String Truncated to 50 Characters with Elipses (...)',
    label: 'String Truncated to 50 Characters with Elipses (...)',
  },
];

/* Consolidation - calculation */
export const calculationValues = [
  {
    value: 'Min',
    label: 'Min',
  }, {
    value: 'Max',
    label: 'Max',
  }, {
    value: 'Avg',
    label: 'Avg',
  }, {
    value: 'Total',
    label: 'Total',
  },
];

/* Formatting table - formats */
export const tableSelectOptions = {
  string: {
    defaultValue: 'Human Readable Model Name',
    options: [{
      value: 'Human Readable Model Name',
      label: 'Human Readable Model Name',
    }, {
      value: 'String Truncated to 50 Characters with Elipses (...)',
      label: 'String Truncated to 50 Characters with Elipses (...)',
    }],
  },
  datetime: {
    defaultValue: 'Date/Time (M/D/Y H:M:S Z)',
    options: [{
      value: 'Date/Time (M/D/Y H:M:S Z)',
      label: 'Date/Time (M/D/Y H:M:S Z)',
    }, {
      value: 'Date (M/D)',
      label: 'Date (M/D)',
    }, {
      value: 'Date (M/D/YY)',
      label: 'Date (M/D/YY)',
    }, {
      value: 'Date (M/D/YYY)',
      label: 'Date (M/D/YYY)',
    }, {
      value: 'Day Full (Monday)',
      label: 'Day Full (Monday)',
    }, {
      value: 'Day Short (Mon)',
      label: 'Day Short (Mon)',
    }, {
      value: 'Day of Month (27)',
      label: 'Day of Month (27)',
    }, {
      value: 'Day of Month (27th)',
      label: 'Day of Month (27th)',
    }, {
      value: 'Day of Week (1)',
      label: 'Day of Week (1)',
    }, {
      value: 'Hour of Day (24)',
      label: 'Hour of Day (24)',
    }, {
      value: 'Month Full',
      label: 'Month Full',
    }, {
      value: 'Year (YYYY)',
      label: 'Year (YYYY)',
    }],
  },
  number: {
    defaultValue: 'Number (1,234)',
    options: [{
      value: 'Number (1,234)',
      label: 'Number (1,234)',
    }, {
      value: 'Number (1,234.0)',
      label: 'Number (1,234.0)',
    }, {
      value: 'Number, 2 Decimals (1,234.00)',
      label: 'Number, 2 Decimals (1,234.00)',
    }, {
      value: 'Convert Numbers Larger than 1.0e+15 to Exponential Form',
      label: 'Convert Numbers Larger than 1.0e+15 to Exponential Form',
    }, {
      value: 'Currency, 2 Decimals ($1,234.00)',
      label: 'Currency, 2 Decimals ($1,234.00)',
    }],
  },
};
