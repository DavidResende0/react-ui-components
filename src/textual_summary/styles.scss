.backgrounded-icon {
  display: inline-block;
  height: 20px;
  width: 20px;
  text-align: center;
  i {
    align-self: center;
    color: #fff;
  }
}
