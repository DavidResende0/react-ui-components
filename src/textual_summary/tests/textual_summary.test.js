import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { TextualSummary } from '../index.jsx';
import { summary1 } from '../data/textual_summary';

describe('TextualSummary', () => {
  it('renders just fine', () => {
    const component = renderer.create(<TextualSummary summary={summary1} onClick={e => null} />);
    const group = component.toJSON();
    expect(group).toMatchSnapshot();
  });
});
