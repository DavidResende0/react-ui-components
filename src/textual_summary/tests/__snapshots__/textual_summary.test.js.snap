// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextualSummary renders just fine 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12 col-lg-6"
  >
    <table
      className="table table-bordered table-hover table-striped table-summary-screen"
    >
      <thead>
        <tr>
          <th
            align="left"
            colSpan="2"
          >
            Properties
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="no-hover"
          onClick={[Function]}
          title={undefined}
        >
          <td
            className="label"
          >
            Region
          </td>
          <td>
            
            <span>
               
              US East (N. Virginia)
            </span>
          </td>
        </tr>
        <tr
          className="no-hover"
          onClick={[Function]}
          title={undefined}
        >
          <td
            className="label"
          >
            Type
          </td>
          <td>
            
            <span>
               
              Amazon EC2 Network
            </span>
          </td>
        </tr>
        <tr
          className="no-hover"
          onClick={[Function]}
          title={undefined}
        >
          <td
            className="label"
          >
            Management Engine GUID
          </td>
          <td>
            
            <span>
               
              null
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="table table-bordered table-hover table-striped table-summary-screen"
    >
      <thead>
        <tr>
          <th
            align="left"
            colSpan="2"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="no-hover"
          onClick={[Function]}
          title="Ok"
        >
          <td
            className="label"
          >
            Default Credentials
          </td>
          <td>
            
            <span>
               
              Valid
            </span>
          </td>
        </tr>
        <tr
          className="no-hover"
          onClick={[Function]}
          title="Ok"
        >
          <td
            className="label"
          >
            SmartState Docker Credentials
          </td>
          <td>
            
            <span>
               
              Valid
            </span>
          </td>
        </tr>
        <tr
          className="no-hover"
          onClick={[Function]}
          title={null}
        >
          <td
            className="label"
          >
            Last Refresh
          </td>
          <td>
            
            <table
              cellPadding="0"
              cellSpacing="0"
            >
              <tbody>
                <tr
                  className="no-hover"
                  onClick={[Function]}
                  title={undefined}
                >
                  <td
                    style={
                      Object {
                        "border": 0,
                        "margin": 0,
                        "padding": 0,
                      }
                    }
                  >
                    
                    Success - 4 Months Ago
                  </td>
                </tr>
                <tr
                  className="no-hover"
                  onClick={[Function]}
                  title={undefined}
                >
                  <td
                    style={
                      Object {
                        "border": 0,
                        "margin": 0,
                        "padding": 0,
                      }
                    }
                  >
                    
                    null
                  </td>
                </tr>
              </tbody>
            </table>
          </td>
        </tr>
        <tr
          className="no-hover"
          onClick={[Function]}
          title={undefined}
        >
          <td
            className="label"
          >
            Last Refresh Date
          </td>
          <td>
            
            <span>
               
              2018-01-20T23:35:50.908Z
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="col-md-12 col-lg-6"
  >
    <table
      className="table table-bordered table-hover table-striped table-summary-screen"
    >
      <thead>
        <tr>
          <th
            align="left"
            colSpan="2"
          >
            Relationships
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="no-hover"
          onClick={[Function]}
          title={undefined}
        >
          <td
            className="label"
          >
            Parent Cloud Provider
          </td>
          <td>
            
            <span>
               
              [object Object]
            </span>
          </td>
        </tr>
        <tr
          className="no-hover"
          onClick={[Function]}
          title={undefined}
        >
          <td
            className="label"
          >
            Cloud Tenants
          </td>
          <td>
            <i
              className="pficon pficon-cloud-tenant"
              title={null}
            />
            <span>
               
              0
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
          title="Show all Cloud Networks"
        >
          <td
            className="label"
          >
            Cloud Networks
          </td>
          <td>
            <i
              className="ff ff-cloud-network"
              title="Show all Cloud Networks"
            />
            <span>
               
              1
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
          title="Show all Cloud Subnets"
        >
          <td
            className="label"
          >
            Cloud Subnets
          </td>
          <td>
            <i
              className="pficon pficon-network"
              title="Show all Cloud Subnets"
            />
            <span>
               
              6
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
          title="Show all Network Routers"
        >
          <td
            className="label"
          >
            Network Routers
          </td>
          <td>
            <i
              className="pficon pficon-route"
              title="Show all Network Routers"
            />
            <span>
               
              1
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
          title="Show all Security Groups"
        >
          <td
            className="label"
          >
            Security Groups
          </td>
          <td>
            <i
              className="pficon pficon-cloud-security"
              title="Show all Security Groups"
            />
            <span>
               
              18
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
          title="Show all Floating IPs"
        >
          <td
            className="label"
          >
            Floating IPs
          </td>
          <td>
            <i
              className="ff ff-floating-ip"
              title="Show all Floating IPs"
            />
            <span>
               
              14
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
          title="Show all Network Ports"
        >
          <td
            className="label"
          >
            Network Ports
          </td>
          <td>
            <i
              className="ff ff-network-port"
              title="Show all Network Ports"
            />
            <span>
               
              26
            </span>
          </td>
        </tr>
        <tr
          className=""
          onClick={[Function]}
          title="Show all Load Balancers"
        >
          <td
            className="label"
          >
            Load Balancers
          </td>
          <td>
            <i
              className="ff ff-load-balancer"
              title="Show all Load Balancers"
            />
            <span>
               
              2
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="table table-bordered table-hover table-striped table-summary-screen"
    >
      <thead>
        <tr>
          <th
            align="left"
            colSpan="2"
          >
            Overview
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className=""
          onClick={[Function]}
          title="Show topology"
        >
          <td
            className="label"
          >
            Topology
          </td>
          <td>
            <i
              className="pficon pficon-topology"
              title="Show topology"
            />
            <span>
               
              undefined
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="table table-bordered table-striped table-summary-screen"
    >
      <thead>
        <tr>
          <th
            align="left"
            colSpan="2"
          >
            Smart Management
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="label"
          >
            Managed by Zone
          </td>
          <td>
            <i
              className="pficon pficon-zone"
              title={null}
            />
            Amazon Zone
          </td>
        </tr>
        <tr>
          <td
            className="label"
          >
            My Company Tags
          </td>
          <td>
            <i
              className="fa fa-tag"
              title={null}
            />
            No My Company Tags have been assigned
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
